storage/range_command.go

```diff
@@ -337,21 +340,26 @@ func (r *Range) InternalRangeLookup(batch engine.Engine, args *proto.InternalRan
 			"Range lookup specified invalid maximum range count %d: must be > 0", rangeCount))
 		return nil
 	}
-	if args.IgnoreIntents {
-		rangeCount = 1 // simplify lookup because we may have to retry to read new
+	consistent := args.ReadConsistency != proto.INCONSISTENT
+	if consistent && args.IgnoreIntents {
+		reply.SetGoError(util.Errorf("can not read consistently and skip intents"))
+		return nil
+	}
+	if args.IgnoreIntents || consistent {
```

Comment:
I don't think we want to make the assumption that consistent means no prefetching. I'd remove the `|| consistent` here. The two are really orthogonal.